RefrigeratorExe/RefrigeratorExe/Item.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace RefrigeratorExe
+{
+    internal class Item : IComparable<Item>
+    {
+        private string _type;
```

Comment:
no need to hold private property and public property to the same member.
final members should look like: 

public Guid Id { get; }
public string Name { get; }
public Guid? ShelfId { get; set; }
public string Type { get; }
public string Kosher { get; }
public DateTime ExpiryDate { get; }
public int Space { get; }
private const int MaxItemLength = 20;

and see my comment on constructor code to complete understanding